### DISCUSSION TITLE
Fit selected estimators to chain ladder

### DISCUSSION THREAD

**Discussion by paulinaKr** [casact/chainladder-python#396] [Q&A] [answered]
Hello, 
I am running an analysis, where I estimate development patterns, based on different triangles: incurred, paid or claim count. My development estimator looks like that: 
`dev_est_p = cl.Development().fit(triangles['earned_premium'])`

`dev_est_t = cl.Development().fit(triangles['total_incurred'])`

`dev_est_c = cl.Development().fit(triangles['claim_count'])`

I would like to check the estimated ultimates for total incurred from chain ladder, according to each of these estimators. When I fit the estimators to the chain ladder, all of them give the same result for ultimates. 
`model = cl.Chainladder().fit(dev_est_p.fit_transform(triangles['total_incurred']))`

`model.ultimate_`

I would expect the ultimates to differ according to different development factors. Is there something wrong how I fit the model? Maybe I should use a different model to check how my ultimates change according to different development patterns? 
Thank you, Paulina

**Comment by jbogaardt** [comment, marked as answer]
Hi @paulinaKr ,

Once you fit an estimator, you don't need to fit it again. I think you're implicitly doing that using `fit_transform`. Instead you call `transform` only if you've already fit it:
```python
dev_est_t = cl.Development().fit(triangles['total_incurred']) # This is now fit
model = cl.Chainladder().fit(dev_est_t.transform(triangles['total_incurred'])) # Only calling Development.transform here
```

Alternatively, if you're planning on transforming the same triangle you are fitting, `Pipeline` estimators make the code a lot cleaner and not have to think about `fit` vs `transform`.
```python
pipe = cl.Pipeline(
    [('dev', cl.Development()),
     ('model', cl.Chainladder())] # named this step as "model"
)
pipe.fit(triangles['total_incurred'])
pipe.named_steps.model.ultimate_ # Accessing the ultimate_ from the "model" step
```

**Reply by paulinaKr** [reply]
Hi @jbogaardt. I think I was not clear with what I want to achive. Here is a small example: 
```
# sample data 
d = {'occurrence': ['2000', '2000', '2000', '2001', '2001', '2002'], 'view': ['2000', '2001', '2002', '2001', '2002', '2002'], 
'cost': [200, 380, 140, 1000, 900, 600], 
'count': [3, 4, 3, 5, 4, 3] }
data = pd.DataFrame(d)
```
```
# create traingle
tri = cl.Triangle(
    data, 
    origin='occurrence',
    development='view', 
    columns=['cost', 'count'], 
    cumulative=True 
)
# check development factors
est = cl.Development().fit(tri)
est_cost = est.ldf_['cost'] #12-24: 1.0667, 24-36:0.3684
est_count = est.ldf_['count'] #12-24: 1, 24:36: 0.75
# use development factors from tri['count'] to estimate ultimate cost
pipe = cl.Pipeline(
    [('dev', cl.Development().fit(tri['count'])),
     ('model', cl.Chainladder())] # named this step as "model"
)
pipe.fit(tri['cost'])
pipe.named_steps.model.ultimate_ 
``` 
The development factors I would like to use are from tri['count'], so: 12-24: 1, 24:36: 0.75, which should give me ultimate for 2002 of 450 (600x1x0.75). However the model uses the development factors estimated from the cost triangle. Is there a way to estimate ultimates using the factors from "count triangle"?

Thank you, Paulina

**Reply by jbogaardt** [reply]
Thanks for the complete example. Its very helpful. If you want to select patterns from one triangle and use those patterns on another, you could set the model up this way:
```python
pipe = cl.Pipeline(
    [('dev', cl.Development()),
     ('model', cl.Chainladder())] 
)

pipe.fit(tri['count']) # use development factors from tri['count'] to estimate ultimate cost
pipe.predict(tri['cost']).ultimate_ # predict 'cost' triangle from the fitted 'count' model.
```
